engine/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM python:3.11.3-slim-buster AS base
+FROM python:3.11.4-slim-bookworm AS base
```

Comment:
we we explicitly use python 3.11.2 anywhere else? (ex. CI, oncall-private, etc)

Reply:
@vadimkerr we may want to update these references to `3.11.3` in a follow-up PR:
- https://github.com/search?q=repo%3Agrafana%2Foncall%203.11.3&type=code
- https://github.com/search?q=repo%3Agrafana%2Foncall-private%203.11.3&type=code

Reply:
Good call! I'll update it today